Updating dependencies and refactoring

NikolayPianikov opened the pull request:
- Updating dependencies 
- Refactoring
- Code cleaning
- Migration to C# 12

NikolayPianikov committed:
Update deps

NikolayPianikov committed:
Use nullable annotations

NikolayPianikov committed:
Code cleanup

NikolayPianikov committed:
Code cleanup & C# 12

NikolayPianikov commented:
@mchechulnikov will you look at the PR?

mchechulnikov commented:
@NikolayPianikov could you please briefly describe what was refactored and for what goal?

NikolayPianikov commented:
> @NikolayPianikov could you please briefly describe what was refactored and for what goal?

Yeah, sure, here:
- Updated all dependencies to newer versions
- Moved to C# 12 and Nullable types.
- Removed Nullable attributes
- Cleaned up the code according to Rider's recommendations

NikolayPianikov committed:
Update deps
